### DISCUSSION TITLE
What, other than redraw, causes a component's view function to be called (endlessly)?

### DISCUSSION THREAD

**Discussion by EverettMcKay** [MithrilJS/mithril.js#2956] [General]
I just discovered that for one of my components, the view function is called endlessly. I discovered this by accident by console logging, but otherwise the component is behaving as it should. I have confirmed that this is not normal by doing the same in other components.

What is causing this? I confirmed that m.redraw() is not being called and v.state is modified only in event handlers. What else causes redraws?

**Comment by dead-claudia** [comment]
The only way Mithril redraws is if one of the following occurs:

1. You invoke `m.render` directly (which sidesteps the autoredraw mechanism)
2. You invoke `m.redraw()`
3. You invoke `m.redraw.sync()`
4. A Mithril-managed event handler is invoked and returns a value other than `false`

There are ways to cause render loops without explicit renders, though. For example, you could do it with `focus` and `blur` events:

- On some element A, you add an `onfocus` handler that sets a flag to add another element B, and an `onblur` to remove it.
- You set up code such that if there's no focused element, that element A is focused.
- Element B has code to enforce focus when present.
- You notice that it doesn't work at first, so you decide to hack in a fix to also restore focus whenever element B is removed.

With the above, you'd be stuck in this loop:

1. No element, element A adds focus.
2. Element A's `onfocus` called, element B added.
3. Element B takes focus, element A's `onblur` handler called.
4. Element B removed, no focus. Workaround re-adds focus to element A.
5. Repeat from step 2.

I'd need to see code to not just take shots in the dark, though.
